src/compiler/checker.ts

```diff
@@ -11544,8 +11544,23 @@ namespace ts {
             const declaringClassDeclaration = <ClassLikeDeclaration>getClassLikeDeclarationOfSymbol(declaration.parent.symbol);
             const declaringClass = <InterfaceType>getDeclaredTypeOfSymbol(declaration.parent.symbol);
 
-            // A private or protected constructor can only be instantiated within it's own class
+            // A private or protected constructor can only be instantiated within its own class or a static method of a subclass
             if (!isNodeWithinClass(node, declaringClassDeclaration)) {
+                const containingFunction = getContainingFunction(node);
+                const containingClass = getContainingClass(node);
+                if (containingClass) {
+                    const containingType = getTypeOfNode(containingClass);
+                    const baseTypes = getBaseTypes(<InterfaceType>containingType);
+                    if (baseTypes.length) {
+                        const baseType = baseTypes[0];
+                        if (containingFunction &&
+                            containingFunction.flags & NodeFlags.Static &&

```

Comment:
Oops. I wasn't paying attention and overfit to the example. I see that instance methods should be ok based on your constructor-with-super equivalence example.